### PR TITLE
url and email validation

### PR DESCRIPTION
**Pull request by teallarson**
- Uses validator npm package to validate emails and urls
- Checks emails against array of "illegal" domains

This should eliminate most bots... things like "demo.com", "example.com", etc can still be used, but we'll see if those present an issue and adjust. I've been using this npm package for a while for form validation.

### CODE REVIEW

**Comment by andyjih** [comment]
@tealjulia This looks great! 👍

File: utils/inflectors.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { stringify } from "gray-matter";
+import validator from "validator";
```
Comment (evantahler) [inline utils/inflectors.ts]: Doesn't look to be in use

File: pages/trial.tsx [hunk 2]
```diff
@@ -79,6 +81,9 @@ export default function Trial({ props }) {
 
   const onSubmit = async (data) => {
     setLoading(true);
+
+    validator.isEmail(data.email);
+    validator.isURL(data.companyWebsite);
```
Comment (evantahler) [inline pages/trial.tsx]: I think these don't actually effect the method?

**Review by evantahler — CHANGES_REQUESTED** [review]
Code-wise, this looks great (other than a few things to remove)!

I'm a little worried about the fact that this PR means that people who don't have a 'work' email address can't contact us.  It seems unlikely, but possible?  Either way, maybe if the error message was changed from:

> 
> Please enter a valid work email.

to

> Please enter a valid work email.  If you are still having trouble you can reach us by email at hello@grouparoo.com

so at least we give folks /some/ way of getting in touch. 
cc @andyjih

**Comment by andyjih** [comment]
> Code-wise, this looks great (other than a few things to remove)!
> 
> I'm a little worried about the fact that this PR means that people who don't have a 'work' email address can't contact us. It seems unlikely, but possible? Either way, maybe if the error message was changed from:
> 
> > Please enter a valid work email.
> 
> to
> 
> > Please enter a valid work email.  If you are still having trouble you can reach us by email at [hello@grouparoo.com](mailto:hello@grouparoo.com)
> 
> so at least we give folks /some/ way of getting in touch.
> cc @andyjih

I disagree, and I don't think adding the extra messaging is necessary. We also have lots of other ways for them to get in touch if they so choose. (slack community, github, they can find the help@ email elsewhere as well). I'd prefer to keep these forms focused on our target persona e.g. people working at companies.

**Comment by teallarson** [comment]
> > Code-wise, this looks great (other than a few things to remove)!
> > I'm a little worried about the fact that this PR means that people who don't have a 'work' email address can't contact us. It seems unlikely, but possible? Either way, maybe if the error message was changed from:
> > > Please enter a valid work email.
> > 
> > 
> > to
> > > Please enter a valid work email.  If you are still having trouble you can reach us by email at [hello@grouparoo.com](mailto:hello@grouparoo.com)
> > 
> > 
> > so at least we give folks /some/ way of getting in touch.
> > cc @andyjih
> 
> I disagree, and I don't think adding the extra messaging is necessary. We also have lots of other ways for them to get in touch if they so choose. (slack community, github, they can find the help@ email elsewhere as well). I'd prefer to keep these forms focused on our target persona e.g. people working at companies.

I'd wondered about that.  My thought was a generic help line under the heading or at the bottom of the page in a smaller font "If you need assistance with registration, please reach out at hello@grouparoo.com"...?

**Comment by evantahler** [comment]
I like the smaller note at the bottom if @andyjih is OK with it!

**Comment by andyjih** [comment]
Sure, a smaller note at the bottom like what we have on the /meet page works for me. We can use the same copy. "For all other inquires, please email help@grouparoo.com"

**Review by andyjih — APPROVED** [review]
lgtm!

**Review by evantahler — APPROVED** [review]

**Pull request merged**